rpcs3/Emu/Cell/PPUInterpreter.cpp

```diff
@@ -25,6 +25,20 @@
 #pragma GCC diagnostic ignored "-Wold-style-cast"
 #endif
 
+void ppubreak(ppu_thread& ppu)
+{
+	if (!g_breakpoint_handler->IsBreakOnBPM()) return;
+
+	bool status = ppu.state.test_and_set(cpu_flag::dbg_pause);
+#ifdef WITH_GDB_DEBUGGER
+	fxm::get<GDBDebugServer>()->notify();
+#endif
+	if (!status && ppu.check_state())
+	{
+		return;
```

Comment:
is this your way of using the check_state return value?

Reply:
this is a typo.

Reply:
No idea if it is necessary but I'm pretty sure I called check_state to make sure the state change was handled immediately.
I changed the code to make it clearer.